### PR TITLE
schedulers: hot region api is independence from hot region

### PR DESCRIPTION
**Pull request by bufferflies**
Signed-off-by: bufferflies <1045931706@qq.com>

### What problem does this PR solve?
fix issue:#3879

### What is changed and how it works?
create new file to statistics  hot region info 
### Check List

Tests

- Unit test

Code changes

Side effects

Related changes
- Need to cherry-pick to the release branch

### Release note

```release-note
None
```

### CODE REVIEW

File: server/cluster/coordinator.go [hunk 1]
```diff
@@ -503,15 +488,15 @@ func (c *coordinator) collectHotSpotMetrics() {
...
-	collectHotMetrics(s, stores, schedulers.HotReadRegionType)
+	c.collectHotMetrics(stores, schedulers.HotReadRegionType)
 	// Collects pending influence.
 	collectPendingInfluence(s, stores)
 }
 
-func collectHotMetrics(s *scheduleController, stores []*core.StoreInfo, typ string) {
-	status := s.Scheduler.(hasHotStatus).GetHotStatus(typ)
+func (c *coordinator) collectHotMetrics(stores []*core.StoreInfo, typ string) {
+	status := schedulers.GetReadStoresLoad(c.cluster)
```
Comment (rleungx) [inline server/cluster/coordinator.go:499]: What if the type is `write`?

File: server/schedulers/hot_region_statistics.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	"github.com/tikv/pd/server/core"
+	"github.com/tikv/pd/server/schedule/opt"
+	"github.com/tikv/pd/server/statistics"
+)
+
+// GetReadStoresLoad returns read store load
+func GetReadStoresLoad(cluster opt.Cluster) *statistics.StoreHotPeersInfos {
+	followStoresDetails := getStoresLoad(cluster, read, core.RegionKind)
+	leaderStoresDetails := getStoresLoad(cluster, read, core.LeaderKind)
+	asLeader := make(statistics.StoreHotPeersStat, len(followStoresDetails))
```
Comment (rleungx) [inline server/schedulers/hot_region_statistics.go:26]: followStoresDetails?

**Review by lhy1024 — COMMENTED** [review]
plz fix

**Comment by bufferflies** [comment]
ref #4149

**Pull request closed without merging**